stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -33,18 +47,278 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
   }
 
   @ChromeDevtoolsMethod
-  public JsonRpcResult getSupportedCSSProperties(JsonRpcPeer peer, JSONObject params) {
-    GetSupportedCSSPropertiesResponse response = new GetSupportedCSSPropertiesResponse();
-    response.cssProperties = Collections.emptyList();
-    return response;
+  public JsonRpcResult getComputedStyleForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetComputedStyleForNodeRequest request =  mObjectMapper.convertValue(
+      params,
+      GetComputedStyleForNodeRequest.class);
+
+    final ArrayListAccumulator<StyleInfo> styles = new ArrayListAccumulator<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        mDocument.getElementStyle(request.nodeId, styles);
+      }
+    });
+
+    return new GetComputedStyleForNodeResult(styles);
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetMatchedStylesForNodeRequest request = mObjectMapper.convertValue(
+      params,
+      GetMatchedStylesForNodeRequest.class);
+
+    final ArrayListAccumulator<StyleInfo> styles = new ArrayListAccumulator<StyleInfo>() {
+      @Override
+      public void store(StyleInfo object) {
+        if (!object.isDefault()) {
```

Comment:
I'm betting we're going to be filtering out a lot of the `StyleInfo` objects? Many will be the default value? This leads me to think that a custom `Accumulator`, ala `AttributeAccumulator`, will be a better idea here. Then we wouldn't have to allocate unless we're going to transmit. Something like this might work:

```
interface StyleInfoAccumulator {
    void store(String name, String value, boolean isDefault);
}
```

And then right here our custom accumulator implementation would only have to do `new StyleInfo` for when `isDefault` is false.